### PR TITLE
Edit extension settings, and more

### PR DESCRIPTION
**Pull request by Klap-in**
- minor improvement for naming/description cli commands
- improve handling zero english translations
- phpdocs
- add page to show settings, and request a one-time edit url
- add page to edit settings. Only one successful save. Implements #19
- if git clone url or branch modified, delete local repository, so that it can be cloned again.
- improve author parsing, e.g. https://github.com/splitbrain/dokuwiki/pull/2211/files

### CODE REVIEW

**Comment by Klap-in** [comment]
Who can review these editable settings for the Translation Tool?

**Comment by Klap-in** [comment]
@dom-mel or @splitbrain Could you review these editable settings for the Translation Tool? Thanks!

File: src/org/dokuwiki/translatorBundle/Controller/ExtensionController.php [hunk 1]
```diff
@@ -135,4 +138,120 @@ public function showAction($type, $name) {
...
+                return $this->redirect($this->generateUrl('dokuwiki_translator_extension_settings', $param));
+            }
+        }
+
+        $data['repository'] = $repository;
+        $data['form'] = $form->createView();
+        return $this->render('dokuwikiTranslatorBundle:Extension:edit.html.twig', $data);
+    }
+
+    private function updateExtension(RepositoryEntity $repositoryEntity, $originalvalues) {
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Controller/ExtensionController.php]: A doc block might be useful

File: src/org/dokuwiki/translatorBundle/Controller/ExtensionController.php [hunk 2]
```diff
@@ -135,4 +138,120 @@ public function showAction($type, $name) {
...
+        $entityManager = $this->getDoctrine()->getManager();
+        $entityManager->merge($repositoryEntity);
+        $entityManager->flush();
+
+        $changed = $originalvalues['branch'] !== $repositoryEntity->getBranch()
+                || $originalvalues['url'] !== $repositoryEntity->getUrl();
+
+        if($changed) {
+            $repository = $this->get('repository_manager')->getRepository($repositoryEntity);
+            $repository->deleteRepository();
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Controller/ExtensionController.php]: without knowing the code, this seems weird? What exactly is deleted here?

File: src/org/dokuwiki/translatorBundle/Form/RepositoryCreateType.php [hunk 3]
```diff
@@ -9,8 +9,14 @@
 
 class RepositoryCreateType extends AbstractType {
 
+    public static $ACTION_CREATE = 'create';
+    public static $ACTION_EDIT = 'edit';
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Form/RepositoryCreateType.php]: These could be `const`

File: src/org/dokuwiki/translatorBundle/Resources/views/Extension/edit.html.twig [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    <h1>Edit {{ form.vars.data.type|capitalize }} Settings</h1>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+            <div class="span6">
+                <p>
+                    Here you can edit the settings of your {{ form.vars.data.type }}. You can
+                    use this edit URL one time. After a succesfull save of this form you need to
+                    request a new edit URL for eventually other edits.
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Resources/views/Extension/edit.html.twig]: I think eventually is the wrong word here. How about "...for any additional edits"?

File: src/org/dokuwiki/translatorBundle/Resources/views/Extension/settings.html.twig [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                <form action="{{ path('dokuwiki_translator_extension_settings', {'type': repository.type, 'name': repository.name}) }}" class="form-horizontal" method="post">
+                    {{ form_widget(form) }}
+
+                    <div class="controls">
+                        <input type="submit" class="btn btn-primary" value="Request Setting Edit URL"/>
+                        {#{% if repository.type == 'core' %}#}
+                            {#<a href="{{ path("dokuwiki_translator_show") }} " class="btn">Abort</a>#}
+                        {#{% else %}#}
+                            {#<a href="{{ path("dokuwiki_translator_show_extension", {'type': repository.type, 'name': repository.name}) }} " class="btn">Abort</a>#}
+                        {#{% endif %}#}
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Resources/views/Extension/settings.html.twig]: can this commented code be removed?

File: src/org/dokuwiki/translatorBundle/Services/Repository/Repository.php [hunk 6]
```diff
@@ -467,12 +477,20 @@ private function applyChanges(GitRepository $git, $folder, TranslationUpdateEnti
         if (!$changes) throw new NoLanguageFileWrittenException();
     }
 
-    private function delete() {
-        $path = $this->buildBasePath();
+    /**
+     * Deletes the folder with the git repository checkout
+     */
+    public function deleteRepository() {
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Services/Repository/Repository.php]: Ahh that answers my previous question. Maybe `deleteCloneDirectory()` would be a better name?

File: src/org/dokuwiki/translatorBundle/Tests/Services/Language/LanguageFileParserTest.php [hunk 7]
```diff
@@ -337,10 +404,12 @@ function testCompleteFile() {
  * german language file
  *
  * @license    GPL 2 (http://www.gnu.org/licenses/gpl.html)
+ *
+ * @package DokuWiki\lang\de\settings
 ';
 
         $this->assertEquals($expectedheader, $parser->getHeader());
-        $this->assertEquals(18, count($parser->getAuthor()->getAll()));
+        $this->assertEquals(20, count($parser->getAuthor()->getAll()));
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Tests/Services/Language/LanguageFileParserTest.php]: I guess this test doesn't make too much sense as this value will probably change over time? or is it working on a static copy?

File: src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php [hunk 8]
```diff
@@ -32,7 +32,7 @@ public function testLicenceSurroundedByEmptyCommentlines() {
 <?php
 
 /**
- * @license    GPL 2 (http://www.gnu.org/licenses/gpl.html)
+ * @license    GPL 3 (http://www.gnu.org/licenses/gpl.html)
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php]: No, we need to stay at 2! We might need to change the URL to https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html

File: src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php [hunk 9]
```diff
@@ -85,4 +85,66 @@ public function testNostrings() {
...
 
+    public function testLicencewithmoreAndNoEmails() {
+
+        $expected = <<<'CONTENT'
+<?php
+
+/**
+ * german language file
+ *
+ * @license    GPL 4 (http://www.gnu.org/licenses/gpl.html)
```
Comment (splitbrain) [inline src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php]: ^_^ GPL 4 does not exist

File: src/org/dokuwiki/translatorBundle/Services/Repository/Repository.php (hunk 6 above)
Comment (Klap-in) [reply src/org/dokuwiki/translatorBundle/Services/Repository/Repository.php]: That 'repository' is a quite ambivalent name. Name is updated

File: src/org/dokuwiki/translatorBundle/Tests/Services/Language/LanguageFileParserTest.php (hunk 7 above)
Comment (Klap-in) [reply src/org/dokuwiki/translatorBundle/Tests/Services/Language/LanguageFileParserTest.php]: It parses a static copy, so numbers are directly related to src/org/dokuwiki/translatorBundle/Tests/Services/Language/testLang.php

File: src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php (hunk 8 above)
Comment (Klap-in) [reply src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php]: Sure, this is a unit test. It tests if licenses as defined by language file author are maintained. Maybe this not most handy example..

Comment (splitbrain) [reply src/org/dokuwiki/translatorBundle/Tests/Services/Language/LocalTextTest.php]: Ahh, I missed that this is a test... looked like a template in the diff.

**Comment by Klap-in** [comment]
Thank you very much for your review!

**Comment by splitbrain** [comment]
No, thank you for all the work! :-)

Can you bring this online on the server or do you need anything from me for that?

**Pull request merged**